src/main/java/org/folio/linked/data/integration/event/CreateResourceEventProducer.java

```diff
@@ -0,0 +1,14 @@
+package org.folio.linked.data.integration.event;
+
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import org.folio.linked.data.model.entity.Resource;
+
+public interface CreateResourceEventProducer extends Predicate<Resource>, Consumer<Resource> {
+
+  default void produce(Resource resource) {
```

Comment:
This method is duplicated in the 3 event producers. Should we create a new `ResourceEventProducer` interface and move it there? Or is that lot of optimization?

Reply:
Oops.. Ignore this message. Just noticed that methods have different signature.

Reply:
What if we include operation type into a method as an argument? It will let us to keep only one single interface ResourceEventProducer, which implementations will 'test' a resource with a operationType at once? It will also let us inject only one set of producers to ResourceModificationEventListener.